File: docs/snippets/typescript/cli/basic_cli.ts
```typescript
import { spawn } from "child_process";

interface ExtractionResult {
  content: string;
  format?: string;
  languages?: string[];
}

async function extractWithCli(
  filePath: string,
  outputFormat: string = "text"
): Promise<string | ExtractionResult> {
  return new Promise((resolve, reject) => {
    const child = spawn("kreuzberg", ["extract", filePath, "--format", outputFormat]);

    let stdout = "";
    let stderr = "";

    child.stdout.on("data", (data) => {
      stdout += data.toString();
    });

    child.stderr.on("data", (data) => {
      stderr += data.toString();
    });

    child.on("close", (code) => {
      if (code !== 0) {
        reject(new Error(`CLI exited with code ${code}: ${stderr}`));
        return;
      }

      if (outputFormat === "json") {
        resolve(JSON.parse(stdout));
      } else {
        resolve(stdout);
      }
    });
  });
}

const document = "document.pdf";
const textOutput = await extractWithCli(document, "text");
console.log(`Extracted: ${(textOutput as string).length} characters`);

const jsonOutput = (await extractWithCli(document, "json")) as ExtractionResult;
console.log(`Format: ${jsonOutput.format}`);
```

File: docs/snippets/python/cli/basic_cli.py
```python
import subprocess
import json
import sys
from pathlib import Path

def extract_with_cli(file_path: str, output_format: str = "text") -> str:
    result = subprocess.run(
        ["kreuzberg", "extract", file_path, "--format", output_format],
        capture_output=True,
        text=True,
    )

    if result.returncode != 0:
        print(f"Error: {result.stderr}", file=sys.stderr)
        sys.exit(1)

    if output_format == "json":
        return json.loads(result.stdout)
    return result.stdout

document = "document.pdf"
text_output = extract_with_cli(document, "text")
print(f"Extracted: {len(text_output)} characters")

json_output = extract_with_cli(document, "json")
print(f"Format: {json_output.get('format', 'unknown')}")
```
